src/components/NewClusterModal.js

```diff
@@ -211,27 +211,35 @@ export const NewClusterModal = withModalDrawer({ width: 675 })(class NewClusterM
     return { size: newPersistentDisk.size }
   }
 
-  sendDeleteMetrics(isDeleteRuntime, isDeletePersistentDisk) {
-    Ajax().Metrics.captureEvent(Events.cloudEnvironmentDelete, {
-      ...extractWorkspaceDetails(this.makeWorkspaceObj()),
-      isDeleteRuntime,
-      isDeletePersistentDisk
-    })
-  }
-
-  sendUpdateMetrics(newRuntime, newPersistentDisk) {
-    const machineType = newRuntime.cloudService === cloudServices.GCE ? newRuntime.machineType : newRuntime.masterMachineType
-    const { cpu, memory } = findMachineType(machineType)
+  sendCloudEnvironmentMetrics() {
+    const metricsEvent = Utils.cond(
+      [(this.state.viewMode === 'deleteEnvironmentOptions'), () => 'cloudEnvironmentDelete'],
+      [(!!this.getOldEnvironmentConfig().runtime), () => 'cloudEnvironmentUpdate'],
+      () => 'cloudEnvironmentCreate'
+    )
+    const { runtime: newRuntime, persistentDisk: newPersistentDisk } = this.getNewEnvironmentConfig()
+    const { runtime: oldRuntime, persistentDisk: oldPersistentDisk } = this.getOldEnvironmentConfig()
```

Comment:
You can move this destructuring above the cond to use the destructuring in the cond.